site.go

```diff
@@ -182,14 +185,26 @@ func (s *Site) GetS3Photo(key string) *S3Photo {
 	}
 }
 
-func (s *Site) GetImgixPhoto(key string) *ImgixPhoto {
+func (s *Site) GetScaledPhoto(resizingService string, resizingServiceSecret string, key string) Renderable {
 	if baseUrl, err := url.Parse(s.BaseUrl); err != nil {
 		fmt.Printf("Error trying to parse site base URL. Error: %s\n", err.Error())
 		return nil
 	} else {
-		return &ImgixPhoto{
-			key,
-			baseUrl,
+		if resizingService == "imgix" {
```

Comment:
I'd very much prefer (from my OOP reflexes) not to return a different type of these each and every time. halp?

Reply:
I'm a bit confused about what you want here.

Reply:
i'll have a think about it tomorrow, but my issue is mainly that each time you add in a new 'backend' of sorts, you also have to add in a new 'if' block, which i'm not a fan of.

Reply:
Of the top of my head, I can't think of a better way to handle this. But I'm sure others have solved this problem before.

Let's think a bit and see if we can find a better way. If not, it's good enough for now and works, so we can make do with it.

Reply:
yeah i think i'm thinking pythonically where i can just put `photo.ImgixRescaledPhoto` in the config and the system will know where to look for the appropriate type. i'll leave this for now.